internal/service/ec2/find.go

```diff
@@ -6539,3 +6540,28 @@ func FindNetworkPerformanceMetricSubscriptionByFourPartKey(ctx context.Context,
 
 	return nil, &resource.NotFoundError{}
 }
+
+func FindInstanceStateById(ctx context.Context, conn *ec2.EC2, id string) (*ec2.InstanceState, error) {
+	in := &ec2.DescribeInstancesInput{
+		InstanceIds: aws.StringSlice([]string{id}),
+	}
+
+	out, err := conn.DescribeInstancesWithContext(ctx, in)
+
+	if tfawserr.ErrCodeEquals(err, lightsail.ErrCodeNotFoundException) {
```

Comment:
What are your thoughts on defining the error constant instead of importing the Lightsail service domain for the exception code? If I see error constants missing from the SDK I would normally create them in a service located errors.go file. Here is an example:
https://github.com/hashicorp/terraform-provider-aws/blob/main/internal/service/storagegateway/errors.go

Reply:
Good Catch, This was me using some of my previous work as an example and forgetting to update that part. I also found that there is an API specifically for obtaining instance state instead of getting all instance data. 

I have updated the find function to use this Specific API for state and also removed the need for the lightsail import by using the ec2 error constants. 

I have also added a check to make sure it will not import or attempt to manage terminated instances. 

What are your thoughts?